### ISSUE TITLE
Connection between Windows and MacOS, but not working

### ISSUE THREAD

**Issue by nbourre** [debauchee/barrier#1569]
### What happened?

I have managed to connect my MacOS server with a Windows 10 client. I see that they get connected but the mouse does not travel to the Windows monitor.

I have tried the reverse, but MacOS as a client kept being in the "Barrier is starting" state.

### Version

v2.4.0

### If applicable, where did you install Barrier from?

Downloaded from GitHub

### What OSes are you seeing the problem on? (Check all that apply)

Windows, macOS

### What OS versions are you using?

Windows 10 Pro 21H2 (Build 19044.1526)
MacOS Monterey (Version 12.0.1)

### Relevant log output

```shell
[2022-02-15T14:34:42] INFO: starting server
[2022-02-15T14:34:42] INFO: config file: /private/var/folders/lb/hq1157px7h73_45gyzmw2kch0000gn/T/Barrier.fRspaz
[2022-02-15T14:34:42] INFO: log level: INFO
[2022-02-15T14:34:42] INFO: drag and drop enabled
started server (IPv4/IPv6), waiting for clients
2022-02-15 14:34:42.715953-0500 barriers[92551:1656750] starting cocoa loop
[2022-02-15T14:34:42] INFO: OpenSSL 3.0.0 7 sep 2021
[2022-02-15T14:34:42] ERROR: ssl certificate doesn't exist: /Users/nbourre/Library/Application Support/barrier/SSL/Barrier.pem
[2022-02-15T14:34:48] INFO: OpenSSL 3.0.0 7 sep 2021
[2022-02-15T14:34:48] ERROR: ssl certificate doesn't exist: /Users/nbourre/Library/Application Support/barrier/SSL/Barrier.pem
[2022-02-15T14:34:58] INFO: zeroconf client detected: INFO-PORT03
[2022-02-15T14:34:59] INFO: OpenSSL 3.0.0 7 sep 2021
[2022-02-15T14:34:59] ERROR: ssl certificate doesn't exist: /Users/nbourre/Library/Application Support/barrier/SSL/Barrier.pem
[2022-02-15T14:35:02] INFO: OpenSSL 3.0.0 7 sep 2021
[2022-02-15T14:35:02] ERROR: ssl certificate doesn't exist: /Users/nbourre/Library/Application Support/barrier/SSL/Barrier.pem
[2022-02-15T14:35:18] INFO: OpenSSL 3.0.0 7 sep 2021
[2022-02-15T14:35:18] ERROR: ssl certificate doesn't exist: /Users/nbourre/Library/Application Support/barrier/SSL/Barrier.pem
[2022-02-15T14:35:34] INFO: OpenSSL 3.0.0 7 sep 2021
[2022-02-15T14:35:34] ERROR: ssl certificate doesn't exist: /Users/nbourre/Library/Application Support/barrier/SSL/Barrier.pem
```

**Comment by jasonmccallister** [comment]
I'm also seeing the same thing, using macOS as the server and a linux desktop as the client. The macOS is throwing the error about the certificate:

```
ERROR: ssl certificate doesn't exist: /Users/<username>/Library/Application Support/barrier/SSL/Barrier.pem
```

I installed Barrier on macOS through brew, trying a manual install now to see if that resolves it.

**Comment by kaunix** [comment]
Dogpyle same thing. newest version of macOS as a server and linux mint as a client.

**Comment by pzoltowski** [comment]
doesn't work for me either. Using between MacOS Big Sur and Windows 10 (surface go). Can connect interactively but mouse sharing doesn't work no matter if macos or windows is the server

**Comment by kaunix** [comment]
in my case it turned out to be a screen alignment issue.
I have two monitors (a 50inch 4k tv and an imac)
if I don't align the imac so that it's just a little farther to the right than the tv the tv will be the only screen I can switch computers from.

**Comment by mrnarayanan** [comment]
I'm getting the same error regarding the ssl certificate. I installed through github. Windows 10 (desktop) as server and MacBook Pro (2020 M1) as client.

**Comment by 8fdafs2** [comment]
Same here. Windows 10 21H2 as the server and macOS Monterey 12.2 as the client, no matter SSL is on or off, the cursor disappears on both desktops.

**Comment by marshall** [comment]
I was able to fix this by generating the `Barrier.pem` and fingerprint manually on my windows server using WSL2 (thanks to https://github.com/debauchee/barrier/issues/1563 for the suggestion):

```shell
$ cd /mnt/c/Users/<user>/AppData/Local/Barrier/SSL/
$ openssl req -x509 -nodes -days 365 -subj /CN=Barrier -newkey rsa:4096 -keyout Barrier.pem -out Barrier.pem
$ mkdir Fingerprints
$ openssl x509 -fingerprint -sha1 -noout -in Barrier.pem > Fingerprints/Local.txt
```

**Comment by mrnarayanan** [comment]
> I'm getting the same error regarding the ssl certificate. I installed through github. Windows 10 (desktop) as server and MacBook Pro (2020 M1) as client.

It was an ssl issue for me. I went back to version 2.3.4 and it worked properly. I believe disabling ssl on 2.4 would also work.

**Comment by 8fdafs2** [comment]
@marshall @mrnarayanan
The problem here seems to be related to the DPI scaling of Windows (I use 200% for a 4k display) and finally gets solved just now by following instructions of  https://github.com/debauchee/barrier/issues/42#issuecomment-396000618, but on barriers.exe instead of barrier.exe.

It's probably not an SSL credential issue in my case.

**Comment by mdjahidhasan009** [comment]
Thanks @marshall its works

**Comment by kennyhyun** [comment]
> ```shell
> $ cd /mnt/c/Users/<user>/AppData/Local/Barrier/SSL/
> $ openssl req -x509 -nodes -days 365 -subj /CN=Barrier -newkey rsa:4096 -keyout Barrier.pem -out Barrier.pem
> $ mkdir Fingerprints
> $ openssl x509 -fingerprint -sha1 -noout -in Barrier.pem > Fingerprints/Local.txt
> ```

Thanks @marshall
It was working for my Mac host.
Started with `cd /Users/<user>/Library/Application Support/barrier/SSL/`

**Comment by nbourre** [comment]
Using this solution from [Marshall](https://github.com/debauchee/barrier/issues/1569#issuecomment-1057341877)

In Windows, for those having a problem with the incorrect name related to `name expected to be in format...` use `//CN=Barrier` 

For those who don't have WSL2 configured, you can also use Git Bash.

**Comment by hueyyeng** [comment]
> ```shell
> $ cd /mnt/c/Users/<user>/AppData/Local/Barrier/SSL/
> $ openssl req -x509 -nodes -days 365 -subj /CN=Barrier -newkey rsa:4096 -keyout Barrier.pem -out Barrier.pem
> ```

Thanks @marshall and also @OmarShawky1 for the commands. Although my situation is slightly different where after generating the `Barrier.pem` file, the Barrier client prompted a dialog to trust the fingerprint and after accepting it, it works. I didn't need to run the command to output the `Fingerprints/Local.txt`.

My setup is Windows 10 (Build 19044) being the Barrier Server and the MBP 2021 model running macOS Monterey (12.3.1) as the Barrier Client. Both using Barrier v2.4.0.

**Comment by kaunix** [comment]
Do you have a terminal window open on the Mac? If so close it

**Comment by hueyyeng** [comment]
@kaunix I did have iTerm2 running but after generating the pem file, Barrier works without any issue.

**Comment by kaunix** [comment]
Yeah for some reason imterm2 works but the native terminal blocks the keyboard

**Comment by Quadstronaut** [comment]
> Yeah for some reason imterm2 works but the native terminal blocks the keyboard
> […](#)
> On May 29, 2022, at 02:09, Huey Yeng ***@***.***> wrote: ﻿ @kaunix I did have iTerm2 running but after generating the pem file, Barrier works without any issue. — Reply to this email directly, view it on GitHub, or unsubscribe. You are receiving this because you were mentioned.

I am not having that issue. I have Terminal open right now and keyboard working fine on both server and client. Latest Barrier, MacOS 11.6.8 (20G730).

**Comment by grtmaddy00** [comment]
For those who are struggling between mac and windows, I found a hack. 
Environment: Win 10 (server) Mac Monterey (client)

Earlier I tried the default steps mentioned in the github description but always got connection timed out error. Tried all the approaches for multiple threads but nothing worked.
Then I switched the server to Mac and client to Windows. With this I was able to mac keyboard and trackpad for Windows but I wanted vice versa. Now here again I switched the client and server to the original desired configuration and it started working.

Not sure if this will work for you or not, but worth giving a shot

**Comment by grtmaddy00** [comment]
Unfortunately this hack stopped working after restarting Windows. Still looking for a solution here

**Comment by aswathselvam** [comment]
> > ```shell
> > $ cd /mnt/c/Users/<user>/AppData/Local/Barrier/SSL/
> > $ openssl req -x509 -nodes -days 365 -subj /CN=Barrier -newkey rsa:4096 -keyout Barrier.pem -out Barrier.pem
> > $ mkdir Fingerprints
> > $ openssl x509 -fingerprint -sha1 -noout -in Barrier.pem > Fingerprints/Local.txt
> > ```
> 
> Thanks @marshall It was working for my Mac host. Started with `cd /Users/<user>/Library/Application Support/barrier/SSL/`

**Comment by Quadstronaut** [comment]
Just touching this issue - I encounter the same error on latest at the time of writing, and have disabled ssl to ensure connectivity. Fortunately my network is entirely private to me.
